sqlmesh/core/model/definition.py

```diff
@@ -1876,6 +1880,28 @@ def _split_sql_model_statements(
     return query, expressions[:pos], expressions[pos + 1 :]
 
 
+def _resolve_custom_session_properties(
+    defaults: t.Optional[t.Dict[str, t.Any]],
+    provided: t.Optional[exp.Expression] | t.Optional[t.Dict[str, t.Any]],
+) -> t.Optional[exp.Expression]:
+    if provided is not None and isinstance(provided, dict):
```

Comment:
I'm pretty sure you just need `isinstance(provided, dict)` since `isinstance(None, dict)` returns false. Let me know if I'm wrong.

Reply:
Indeed, I thought it might check for an empty dict as well, though that won't cause any issue as it resolved to an empty dict anyway.